_rules/image-filename-as-accessible-name-9eb3f6.md

```diff
@@ -28,26 +28,34 @@ input_aspects:
 acknowledgments:
   authors:
     - Bryn Anderson
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'alt-require'
 ---
 
 ## Applicability
 
-The rule applies to any HTML element with the [semantic role][] of `img` or any HTML `input` element with a [`type`][type] of `image` when each of the following is true:
+The rule applies to any HTML `img` element or any HTML `input` element in the [Image Button][] state when each of the following is true:
 
 - the image is [included in the accessibility tree][]; and
-- the image has an [accessible name][] that is equivalent to the [filename][] specified in the `src` attribute. Difference in letter casing, leading and trailing [whitespace][] should be ignored.
+- the image has a non-empty [accessible name][] that is equivalent to the [filename][] of at least one of the [image sources][] in its [source set][] (for `img` elements), or to the [filename][] specified in its `src` attribute (for [Image Buttons][image button]). Difference in letter casing, leading and trailing [whitespace][] should be ignored.
```

Comment:
I agree that it doesn't read well…

I tried to use the orthogonal enumeration. I am not sure this read much better, notably because the expectation still has a part specific to source sets.


The cleanest way to split this would be to make separate rules. It made sense to group them when we were looking at the same thing (the `src` attribute), but now it make little sense to have them together (as we see with the wonky formulations we have to use).